nni/nas/oneshot/pytorch/supermodule/operation.py

```diff
@@ -673,34 +710,88 @@ def forward_with_args(
             k_proj = _S(k_proj)[:, :_W(kdim)]
             v_proj = _S(cast(Tensor, self.v_proj_weight))[:embed_dim_]
             v_proj = _S(v_proj)[:, :_W(vdim)]
+        else:
+            q_proj = k_proj = v_proj = None
+
+        if kwargs.get("mapping", False):
+            return {
+                "in_proj_bias": in_proj_bias, "in_proj_weight": in_proj_weight,
+                "bias_k": bias_k, "bias_v": bias_v,
+                "out_proj.weight": out_proj_weight, "out_proj.bias": out_proj_bias,
+                "q_proj_weight": q_proj, "k_proj_weight": k_proj, "v_proj_weight": v_proj
+            }
+        else:
+            return in_proj_bias, in_proj_weight, bias_k, bias_v, out_proj_weight, out_proj_bias, q_proj, k_proj, v_proj, qkv_same_embed_dim
+
+    def _save_param_buff_to_state_dict(self, destination, prefix, keep_vars):
```

Comment:
Better to have a dedicated test in `test_oneshot_supermodules`, as it looks over-complicated to me.